shellmetrics

```diff
@@ -151,15 +151,23 @@ pretty() {
   while [ "$i" -lt "$1" ] && i=$((i + 1)); do
     echo 'shellmetrics_wrapper() {'
   done
-  sed "s/\([^\$'\"()]\)\( *()\)/\1_L$MARK\2/g" \
-  | awk "{i++; gsub(/$MARK/,i); print}"
+  mark "$MARK" | awk "{i++; gsub(/$MARK/,i); print}"
   while [ "$i" -gt 0 ] && i=$((i - 1)); do
     echo '}'
   done
   echo "typeset -fp shellmetrics_wrapper 2>/dev/null" \
     "|| typeset -f shellmetrics_wrapper"
 }
 
+mark() {
+  sed " \
+    s/\([^[:space:]\$'\"();&|<>=@!*?+]\)\( *()\)/\1_L${1}\2/g; \
+    s/^[[:space:]]*function[[:space:]]\{1,\}[^[:space:]\$'\"();&|<>]\{1,\}/&_L${1}/g; \
+    s/[;&|(][[:space:]]*function[[:space:]]\{1,\}[^[:space:]\$'\"();&|<>]\{1,\}/&_L${1}/g; \
+    s/_L${1}_L${1}/_L${1}/g \
```

Comment:
`t` を使用すれば、この行は減らせそうですがどうでしょうか？

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html

>[2addr]t [label]
Test. Branch to the : command verb bearing the label if any substitutions have been made since the most recent reading of an input line or execution of a t. If label is not specified, branch to the end of the script.

Reply:
ありがとうございます。

意地悪な例になりますが、以下のように1行に2種類の関数宣言が混在している場合を考えると、両方の形式について置換を実行しておきたいというように考えています (意図を勘違いしていましたらすみません)。
```bash
f() { :; };function g { :; }
```

Reply:
189f7c5 テストケース追加しました。

Reply:
なるほど。たしかにそうですね。~これもテストに加えて頂けると助かります。~ （と書こうとしたら先に書かれていた。）